File: Lab5C++/TVector3.cpp
#include "TVector3.h"
#include <math.h>

TVector3::TVector3(float x, float y, float z)
{
	this->x = x;
	this->y = y;
	this->z = z;
}
float TVector3::get_coord(char coord)
{
	switch (coord)
	{
	case 'x':
		return x;
		break;
	case 'y':
		return y;
		break;
	case 'z':
		return z;
		break;
	}
}
float TVector3::get_length()
{
	return sqrt(x * x + y * y + z * z);
}
float TVector3::scalar(TVector& other)
{
	return x * other.get_coord('x') + y * other.get_coord('y') + z * other.get_coord('z');
}

File: Lab5Py/TVector3.py
from TVector import TVector
from math import sqrt
from math import pow

class TVector3(TVector):
    def __init__(self, x=0, y=0, z=0):
        self.__x=x 
        self.__y=y
        self.__z=z
    def get_length(self):
        return sqrt(pow(self.__x,2)+pow(self.__y,2)+pow(self.__z,2))
    def get_coord(self, coord):
        if coord=='x':
            return self.__x
        elif coord=='y':
            return self.__y
        else:
            return self.__z
    def scalar(self, other):
        return self.__x*other.get_coord('x')+self.__y*other.get_coord('y')+self.__z*other.get_coord('z')




